core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3204,6 +3205,33 @@ private[spark] object Utils extends Logging {
     }
     files.toSeq
   }
+
+  private def isG1GarbageCollector: Boolean = {
```

Comment:
Can this be a `val`? I don't think it will change
Likewise could you just try to get G1HeapRegionSize once here? if it's not G1GC, just store "None". Then you don't check it each time. Because I think this can't change during the JVM's lifetime.
Then you don't need a utility method below, even

Reply:
ok